mne/io/fiff/tests/test_raw_fiff.py

```diff
@@ -739,6 +739,16 @@ def test_proj(tmpdir):
     raw.apply_proj()
     assert_allclose(raw[:, :][0][:1], raw[0, :][0])
 
+    # Read file again, apply proj, pick all channels one proj did NOT apply to;
+    # then try to delete this proj, which now exclusively refers to channels
+    # which are not present in the data anymore.
+    raw = read_raw_fif(fif_fname).apply_proj()
+    del_proj_idx = 0
+    picks = list(set(raw.ch_names) -
+                 set(raw.info['projs'][del_proj_idx]['data']['col_names']))
+    raw.pick(picks)
+    raw.del_proj(del_proj_idx)
```

Comment:
can you test that del_proj_idx is actually no longer there?

Reply:
Is 09c9a9b sufficient?